desk/app/channels.hoon

```diff
@@ -2132,9 +2132,20 @@
       /groups/(scot %p p.flag)/[q.flag]/v1/group-ui
     =+  .^(group=group-ui:g %gx path)
     ?.  (~(has by channels.group) nest)  ca-core
+    ::  toggle the volume based on permissions
+    =/  =source:activity  [%channel nest flag]
+    ?.  (can-read:ca-perms our.bowl)
+      (send:ca-activity [%adjust source ~] ~)
+    =+  .^(=volume-settings:activity %gx (scry-path %activity /volume-settings/noun))
+    =.  ca-core
+      ::  if we don't have a setting, no-op
+      ?~  setting=(~(get by volume-settings) source)  ca-core
+      ::  if they have a setting that's not mute, retain it otherwise
+      ::  delete setting if it's mute so it defaults
+      ?.  =(setting mute:activity)  ca-core
+      (send:ca-activity ~[[%adjust source ~]])
```

Comment:
Because this will run on any permissions changes not just ones where we've regained/lost permissions, so we don't want to clear any settings they might have if the permissions didn't change.